parser.go

```diff
@@ -422,19 +429,85 @@ func (r *Rule) network(key item, l *lexer) error {
 	}
 	switch key.typ {
 	case itemSourceAddress:
-		r.Source.Nets = append(r.Source.Nets, items...)
+		if validNetworks(items) {
+			r.Source.Nets = append(r.Source.Nets, items...)
+		} else {
+			return fmt.Errorf("some or all source ips are invalid: %v", items)
+		}
 	case itemSourcePort:
-		r.Source.Ports = append(r.Source.Ports, items...)
+		if portsValid(items) {
+			r.Source.Ports = append(r.Source.Ports, items...)
+		} else {
+			return fmt.Errorf("some or all source ports are invalid: %v", items)
+		}
 	case itemDestinationAddress:
-		r.Destination.Nets = append(r.Destination.Nets, items...)
+		if validNetworks(items) {
+			r.Destination.Nets = append(r.Destination.Nets, items...)
+		} else {
+			return fmt.Errorf("some or all destination ips are invalid: %v", items)
+		}
 	case itemDestinationPort:
-		r.Destination.Ports = append(r.Destination.Ports, items...)
+		if portsValid(items) {
+			r.Destination.Ports = append(r.Destination.Ports, items...)
+		} else {
+			return fmt.Errorf("some or all destination ports are invalid: %v", items)
+		}
 	default:
 		panic("item is not a network component")
 	}
 	return nil
 }
 
+// Validate that every item is between 1 and 65535.
+func portsValid(p []string) bool {
+	for _, _port := range p {
```

Comment:
let's omit the leading underscores from variable names. while `_` has a special meaning, `_foo` does not. Leading underscores are not used in the rest of the code base, so this introduces a bit of a stylistic gap.

Reply:
done